tokio/src/sync/semaphore.rs

```diff
@@ -47,7 +47,23 @@ use std::sync::Arc;
 /// }
 /// ```
 ///
-/// Use [`Semaphore::acquire_owned`] to move permits across tasks:
+/// Limit number of incoming requests being handled at the same time, using a task-safe semaphore
+/// via [`Semaphore::acquire_owned`].
+///
+/// Similar to limiting the numer of simultaneous opened files, network handles are a limited resource.
+/// Allowing an unbounded amount of requests to be processed could result in a denial-of-service,
+/// and many other issues.
```

Comment:
Network handles are a good example. I suggest that we update the example to accept on a `TcpListener` in a very similar way to the example in the README of this repository. Except this time, we acquire a permit before accepting a connection, and we drop the permit after handling an incoming connection, inside `tokio::spawn`.

I suggest this because right now, we're not really handling "incoming requests". It's just a simple loop.

Reply:
Got it. I've modified the request handler in the README, and repurposed it here.

Do you think that would be prudent -- or would simply sticking to the basic format that was there before be better? For example, removing the inner loop and just have the old "// do work here".

Reply:
I think the third option seems best. It gives you the basic structure in the main body, while still giving a useful "do work" example -- without cluttering everything. 

Plus we don't have to remember to explicitly drop the socket first this way.

Reply:
Makes sense. At that point, we might even entirely remove the `handle_connection` method from the example.

> Plus we don't have to remember to explicitly drop the socket first this way.

Hmm. It could make sense to do `handle_connection(&mut socket).await` to show explicitly dropping it in the example.

Reply:
Yeah, at this point the details of handling the connection are irrelevant.

Reply:
Removed the implementation of `handle_connection`; and so on.